lib/ui_components/login/login_failed_dialog.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:first_project/extensions/context_extensions/colors.dart';
+import 'package:first_project/extensions/context_extensions/text_styles.dart';
+import 'package:first_project/login_state.dart';
+import 'package:first_project/providers/email_state_provider.dart';
+import 'package:first_project/providers/password_state_provider.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class LoginFailedDialog extends ConsumerWidget {
+  const LoginFailedDialog({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return AlertDialog(
+      backgroundColor: Colors.white,
+      title: Center(
+        child: Text(
+          ErrorState.title,
+          style: context.titleMediumBold,
+        ),
+      ),
+      content: Text(
+        ErrorState.message,
+        style: context.bodyMedium,
+        textAlign: TextAlign.center,
+      ),
+      actions: [
+        Center(
+          child: TextButton(
+            style: OutlinedButton.styleFrom(
+                side: BorderSide(color: context.secondary),
+                foregroundColor: context.secondary,
+                backgroundColor: Colors.transparent,
+                shape: RoundedRectangleBorder(
+                  borderRadius: BorderRadius.circular(8),
+                )),
+            onPressed: () {
+              ref.read(emailStateProvider.notifier).state = '';
+              ref.read(passwordStateProvider.notifier).state = '';
```

Comment:
Why do you do this? You want the email and password texts to be removed from the text fields when the request fails?

Currently,on failure the text remains the same on the UI but the email/password are removed from the state